src/platform/backends/qemu/firewall_config.cpp

```diff
@@ -80,138 +94,140 @@ const QString RETURN{QStringLiteral("RETURN")};
 const QString reject_with{QStringLiteral("--reject-with")};
 const QString icmp_port_unreachable{QStringLiteral("icmp-port-unreachable")};
 
-auto multipass_iptables_comment(const QString& bridge_name)
+auto multipass_firewall_comment(const QString& bridge_name)
 {
     return QString("generated for Multipass network %1").arg(bridge_name);
 }
 
-void add_iptables_rule(const QString& table, const QString& chain, const QStringList& rule, bool append = false)
+void add_firewall_rule(const QString& firewall, const QString& table, const QString& chain, const QStringList& rule,
+                       bool append = false)
 {
     auto process = MP_PROCFACTORY.create_process(
-        iptables, QStringList() << wait << dash_t << table << (append ? append_rule : insert_rule) << chain << rule);
+        firewall, QStringList() << wait << dash_t << table << (append ? append_rule : insert_rule) << chain << rule);
 
     auto exit_state = process->execute();
 
     if (!exit_state.completed_successfully())
-        throw std::runtime_error(
-            fmt::format("Failed to set iptables rule for table {}: {}", table, process->read_all_standard_error()));
+        throw mp::FirewallException(
+            fmt::format("Failed to set firewall rule for table {}: {}", table, process->read_all_standard_error()));
 }
 
-void delete_iptables_rule(const QString& table, const QStringList& chain_and_rule)
+void delete_firewall_rule(const QString& firewall, const QString& table, const QStringList& chain_and_rule)
 {
-    auto args = QStringList() << iptables << wait << dash_t << table << delete_rule << chain_and_rule;
+    auto args = QStringList() << firewall << wait << dash_t << table << delete_rule << chain_and_rule;
 
     auto process =
         MP_PROCFACTORY.create_process(QStringLiteral("sh"), QStringList() << QStringLiteral("-c") << args.join(" "));
 
     auto exit_state = process->execute();
 
     if (!exit_state.completed_successfully())
-        throw std::runtime_error(
-            fmt::format("Failed to delete iptables rule for table {}: {}", table, process->read_all_standard_error()));
+        throw mp::FirewallException(
+            fmt::format("Failed to delete firewall rule for table {}: {}", table, process->read_all_standard_error()));
 }
 
-auto get_iptables_rules(const QString& table)
+auto get_firewall_rules(const QString& firewall, const QString& table)
 {
-    auto process = MP_PROCFACTORY.create_process(iptables, QStringList() << wait << dash_t << table << list_rules);
+    auto process = MP_PROCFACTORY.create_process(firewall, QStringList() << wait << dash_t << table << list_rules);
```

Comment:
This logs multiple noisy warnings that I suppose we're ok with:
```
# Warning: iptables-legacy tables present, use iptables-legacy to see them
```

Should we maybe `bash -c` and `grep -v` on stderr?

Reply:
Hmm, ok, I'll look into that.

Reply:
I wouldn't mind  splitting into a separate PR if you prefer. Note that this might need some minutia to get other stderr and stdout in their respective streams.

Reply:
Yeah, I think it would be better to do this separately.  I'll add a TODO though so we can reference it and try not to forget about it :wink:

Reply:
Ok, added a TODO.